gateway/gateway.go

```diff
@@ -116,6 +116,34 @@ func (i ImmutablePath) IsValid() error {
 
 var _ path.Path = (*ImmutablePath)(nil)
 
+type CarParams struct {
+	Range *DagEntityByteRange
+	Scope DagScope
+}
+
+// DagEntityByteRange describes a range request within a UnixFS file.
+// From and To mostly follow [HTTP Byte Range] Request semantics.
+// From >= 0 and To = nil: Get the file (From, Length)
+// From >= 0 and To >= 0: Get the range (From, To)
+// From >= 0 and To <0: Get the range (From, Length - To)
+// From < 0 and To = nil: Get the file (Length - From, Length)
+// From < 0 and To >= 0: Get the range (Length - From, To)
+// From < 0 and To <0: Get the range (Length - From, Length - To)
+//
+// [HTTP Byte Range]: https://httpwg.org/specs/rfc9110.html#rfc.section.14.1.2
+type DagEntityByteRange struct {
```

Comment:
@aschmahmann why not reuse `ByteRange`? I see the `From` is `uint64` on `ByteRange`. Can we change that and unify both?